packages/patrol_cli/lib/src/features/test/ios_test_backend.dart

```diff
@@ -47,110 +50,182 @@ class IOSAppOptions {
   }
 
   final ProcessManager _processManager;
   final FileSystem _fs;
+  final IOSDeploy _iosDeploy;
   final DisposeScope _disposeScope;
   final Logger _logger;
 
-  Future<void> run({
-    required Device device,
-    required IOSAppOptions options,
-  }) async {
-    final targetName = basename(options.target);
-    final task = _logger
-        .task('Building app for $targetName and running it on ${device.name}');
-
-    final flutterProcess = await _processManager.start(
-      options.toFlutterBuildInvocation(device),
-      runInShell: true,
-    );
-
-    flutterProcess.stdout.listen((rawMsg) {
-      final msg = systemEncoding.decode(rawMsg).trim();
-      _logger.detail('\t$msg');
-    }).disposedBy(_disposeScope);
-
-    flutterProcess.stderr.listen((rawMsg) {
-      final msg = systemEncoding.decode(rawMsg).trim();
-      _logger.err('\t$msg');
-    }).disposedBy(_disposeScope);
-
-    var exitCode = await flutterProcess.exitCode;
-    if (exitCode != 0) {
-      task.fail('Failed to run `flutter build` for $targetName');
-      throw Exception('flutter build exited with code $exitCode');
-    }
+  @override
+  Future<void> build(IOSAppOptions options, Device device) async {
+    await _disposeScope.run((scope) async {
+      final subject = '${options.desc} for ${device.platformDescription}';
+      final task = _logger.task('Building $subject');
 
-    final xcodebuildProcess = await _processManager.start(
-      options.toXcodebuildInvocation(device),
-      runInShell: true,
-      workingDirectory: _fs.currentDirectory.childDirectory('ios').path,
-    );
-
-    xcodebuildProcess.stdout.listen((rawMsg) {
-      final msg = systemEncoding.decode(rawMsg).trim();
-      _logger.detail('\t$msg');
-    }).disposedBy(_disposeScope);
-
-    xcodebuildProcess.stderr.listen((rawMsg) {
-      final msg = systemEncoding.decode(rawMsg).trim();
-      _logger.err('\t$msg');
-    }).disposedBy(_disposeScope);
-
-    exitCode = await xcodebuildProcess.exitCode;
-    if (exitCode == 0) {
-      task.complete('Built and ran app for $targetName on ${device.name}');
-    } else {
-      task.fail('Failed to build/run app for $targetName on ${device.name}');
-      throw Exception('xcodebuild exited with code $exitCode');
-    }
+      Process process;
+
+      // flutter build ios --config-only
+
+      var flutterBuildKilled = false;
+      process = await _processManager.start(
+        options.toFlutterBuildInvocation(device),
+        runInShell: true,
+      );
+      scope.addDispose(() async {
+        process.kill();
+        flutterBuildKilled = true; // `flutter build` has exit code 0 on SIGINT
+      });
+      process.listenStdOut((l) => _logger.detail('\t$l')).disposedBy(scope);
+      process.listenStdErr((l) => _logger.err('\t$l')).disposedBy(scope);
+      var exitCode = await process.exitCode;
+      if (exitCode != 0) {
+        final cause = '`flutter build ios` exited with code $exitCode';
```

Comment:
Do you use grave accents anywhere else in logs to indicate commands? If not, let's stick to the convention

Reply:
I use backticks when I want to show the executable command.

but I also do:

```dart
final cause = 'Gradle build exited with code $exitCode';
```

instead of 


```dart
final cause = '`./gradlew` exited with code $exitCode';
```

I think it's fine to leave it as it is.